backend/src/routes/api/settings.routes.js

```diff
@@ -0,0 +1,26 @@
+import { StatusCodes } from 'http-status-codes';
+import Profile from '../../models/Profile.js';
+import Settings from '../../models/Settings.js';
+import { settingsValidator } from '../../models/Settings.js';
+
+const settingsRoutes = (router) => {
+  router.get('/settings/:id', async (req, res) => {
+    const profile = await Profile.findById(req.params.id);
+    if (!profile) {
+      return res.status(StatusCodes.NOT_FOUND).send();
+    }
+    const settings = await Settings.findById(profile.settings);
+    res.status(StatusCodes.OK).send(settings);
+  });
+
+  router.patch('/settings/:id', settingsValidator, async (req, res) => {
+    const profile = await Profile.findById(req.params.id);
+    if (!profile) {
+      return res.status(StatusCodes.NOT_FOUND).send();
+    }
+    const settings = await Settings.findByIdAndUpdate(profile.settings, req.body, { new: true });
+    res.status(StatusCodes.OK).send(settings);
```

Comment:
Applies for both functions:

1. extract body of the middleware to the separate function or even separate file (middlewares directory)
2. use our error handling middleware to deal with errors (instead of sending error responses in lines 10, 19)
3. It is possible that `Settings.findById` throws an error (please handle it as well)